datasets/ONS-UK-trade-in-services/main.py

```diff
@@ -1,58 +1,196 @@
-tables.append(new_table)
+# %%
+trace.add_column("Renaming OBS column into Value and DATAMARKER column into Marker")
+new_table.rename(columns = {'OBS':'Value', 'DATAMARKER':'Marker'},inplace = True)
+new_table
 
+# %%
+new_table['Value'] = pd.to_numeric(new_table['Value'], errors = 'coerce')
 
-# We just combine these two into the same table for now.
 
 # %%
-from pathlib import Path
-out = Path('out')
-out.mkdir(exist_ok=True)
-tidy = pd.concat(tables).drop_duplicates()
+def left(s, amount):
+    return s[:amount]
+def right(s, amount):
+    return s[-amount:]
+def date_time (date):
+    if len(date)  == 5:
+        return 'year/' + left(date, 4)
+    #year/2019
+    elif len(date) == 6:
+        return 'quarter/' + left(date,4) + '-' + right(date,2)
+    #quarter/2019-01
+    else:
+        return date
 
-tidy.to_csv(out / 'observations.csv', index = False)
 
+# %%
+new_table['Period'] = new_table['Period'].astype(str).replace('\.', '', regex=True)
+new_table['Period'] =  new_table["Period"].apply(date_time)
+trace.Period("Formating to be year/0000 and quarter/2019-01 ")
 
-# The metadata is more problematic; for now we just use the latter dataset's metadata.
-# 
-# Todo: review titles, etc.
+# %%
+new_table = new_table.replace({'Marker' : {'-' : 'itis-nil', '..' : 'disclosive'}})
+trace.Marker("Replcaing - to mean itis-nil and .. to mean disclosive")
 
 # %%
-from gssutils.metadata import THEME
-scraper.dataset.theme = THEME['business-industry-trade-energy']
-scraper.dataset.family = 'Trade'
-with open(out / 'observations.csv-metadata.trig', 'wb') as metadata:
-     metadata.write(scraper.generate_trig())
-csvw = CSVWMetadata('https://gss-cogs.github.io/family-trade/reference/')
-csvw.create(out / 'observations.csv', out / 'observations.csv-schema.json')
+new_table['Seasonal Adjustment'] =  'NSA'
+trace.Seasonal_Adjustment("Adding in column Seasonal Adjustment with value NSA")
 
+# %%
+new_table['Flow'] = new_table['Flow'].apply(pathify)
 
-# Generate alternative output for direct CSVW transformation. Note that these files won't be included in the upload to PMD.
+# %%
+tidy = new_table[['ONS Partner Geography', 'Period','Flow','Trade Services', 'Seasonal Adjustment', 'Value', 'Marker' ]]
+
+# %%
+tidy = tidy.drop_duplicates()
+tidy
 
 # %%
-#tidy['Measure Type'] = tidy['Measure Type'].astype('category')
-#tidy['Measure Type'].cat.categories = tidy['Measure Type'].cat.categories.map(
-    #lambda x: pathify(x).replace('-', '_'))
-#tidy.to_csv(out / 'observations-alt.csv', index = False)
-#csvw.create(out / 'observations-alt.csv', out / 'observations-alt.csv-metadata.json', with_transform=True,
-            #base_url='http://gss-data.org.uk/data/', base_path='gss_data/trade/ons-uk-trade-in-services',
-            #dataset_metadata=scraper.dataset.as_quads())
+cubes.add_cube(scraper_1, tidy, "ONS UK trade in services by country" )
+cubes.add_cube(scraper_2, tidy, "ONS UK trade in services by partner country")
+cubes.output_all()
```

Comment:
Both transforms are now within the one data frame named _tidy_, therefore, do not need to repeat the add_cube function. I would replace lines 176-178 with the following: 

```python
cubes.add_cube(scraper, tidy, "ONS UK trade in services by country and partner country" )
cubes.output_all()
```

Reply:
Repeated add_cube function is removed